Sources/Apollo/RecordSet.swift

```diff
@@ -1,28 +1,37 @@
+import Foundation
+
+/// A row of data that contains a `Record` and some associated metadata.
+public struct RecordRow {
+  public internal(set) var record: Record
+  public internal(set) var lastReceivedAt: Date
```

Comment:
Is it fine grained enough to keep this per record or do we need the ability to update this on a per field basis? If a query only fetches a subset of the fields, we still need to know that the fields that weren't fetched can be stale (their `lastUpdated` hasn't changed).